packages/logger/src/middleware/middy.ts

```diff
@@ -30,34 +30,29 @@ import { HandlerOptions, LogAttributes } from '../types';
 const injectLambdaContext = (target: Logger | Logger[], options?: HandlerOptions): middy.MiddlewareObj => {
 
   const loggers = target instanceof Array ? target : [target];
-  const persistentAttributes: LogAttributes[] = [];
+  const persistentAttributes: (LogAttributes | undefined)[] = [];
 
   const injectLambdaContextBefore = async (request: middy.Request): Promise<void> => {
     loggers.forEach((logger: Logger) => {
-      logger.addContext(request.context);
+      const initialPersistentAttributes = Logger.injectLambdaContextBefore(logger, request.event, request.context, options);
```

Comment:
(Optional) the same as above.

Q: I assume that decorator + manual has already been covered by the change in `Logger.ts`?

Reply:
Yep, this is the reason i created the static methods injectLambdaContextBefore and injectLambdaContextAfterAndOnError which I use with middy and the decorator: because I wanted to make sure the behaviour is the same across different instrumentation strategies.